core/block_validator.go

```diff
@@ -137,6 +140,52 @@ func (v *BlockValidator) ValidateBody(block *types.Block) error {
 	return nil
 }
 
+// ValidateSystemTxs validates all system txs contained in a block.
+// We check that:
+// - the sender is a whitelisted address
+// - the recipient is a system contract
+func (v *BlockValidator) ValidateSystemTxs(block *types.Block) error {
+	signer := types.MakeSigner(v.config, block.Number())
+
+	for _, tx := range block.Transactions() {
+		if !tx.IsSystemTx() {
+			continue
+		}
+
+		from, err := types.Sender(signer, tx)
+		if err != nil {
+			return err
+		}
+
+		found := false
+		for _, signer := range v.config.Scroll.SystemTxConfig.Signers {
```

Comment:
As we discussed, there are two ways to authorize system txs:
1. Have a whitelist, and whitelisted senders pass the tx into the txpool like normal transactions. Updating this list is problematic, and the risk of leaking this private key puts the protocol at risk.
2. Like we do with `L1MessageTx`, we disable it in [txpool](https://github.com/scroll-tech/go-ethereum/blob/develop/core/tx_pool.go#L591) and [p2p](https://github.com/scroll-tech/go-ethereum/blob/85447ec9a484a5f2fdc893d2510b5937bd52d8cb/eth/handler.go#L483), and the sequencer produces it [internally](https://github.com/scroll-tech/go-ethereum/blob/85447ec9a484a5f2fdc893d2510b5937bd52d8cb/miner/worker.go#L1427).

I think (2) is better, but open to discussion on this matter.

Reply:
(2) done.